### ISSUE TITLE
"Toggle menu bar" as plugin for Geany

### ISSUE THREAD

**Issue by ralf3u** [xiota/geanylua-scripts#13]
Toggle menu bar of geanylua works absolutely perfect and is so useful for me. So what about the idea to publish `Toggle menu bar` as plugin name for Geany? I think it would be good to offer no options for the plugin. That should make the plugin very simple to code. So, there should be not an option with a checkbox like: `The menu bar will always appear when starting Geany`. If users want options then they should take the geanylua-version. Something like the last sentence could be mentioned inside the plugin in a readme.txt-document. In the readme.txt-document there could be also some text like this: `In case the shortcut to toggle the menu bar was forgotten, press Ctrl+Alt+P to let appear the Prefrences, and then click on the left side on "Keybindings" to search at the end of the list for "Toggle menu bar" to see which shortcut was set. An other way would be to open the document keybindings.conf that is loacated at .config/geany to search there for toggle_menu_bar to see which shortcut was set.` The plugin would be user-friendly and also developer-friendly because everyone knows by its name what the plugin does. Another advantage is that once the plugin runs and is implemented in the official geany plugins there is no need to maintain the plugin.

In issue number 633 at geany, you wrote:
> I already did write a plugin (1138) that addresses this and some other open issues. It was ignored for six months. [...] So I closed it for lack of interest.

Well, I'm sure that there is an interest because you wrote also at geany in 633:
>The only agreement so far is that this feature is better implemented as a plugin.

I think 1138 was ingnored because there were too many things inside the plugin.

Today eth16 wrote a comment at geany-plugins 1380 with the title `[Plugin Request] Toggle menu bar`. And his words sound good to me:
>Such a plugin already exist (though seems abandoned) as mentioned in 633. This could probably be used as a base.

If the plugin `Toggle menu bar` will exist, the next step could be to create the next plugin `Tab title styles`.

What do you think about the plugin `Toggle menu bar`?

**Comment by xiota** [comment]
Are you suggesting... add the lua script to the example plugins included with geanylua plugin?  Or writing an entirely new plugin in C, C++, or other compiled language?

**Comment by ralf3u** [comment]
>Are you suggesting... add the lua script to the example plugins included with geanylua plugin? Or writing an entirely new plugin in C, C++, or other compiled language?

I was thinking about a complete new plugin that would be listed as single plugin in the list of the "Plugin Manager" with the name `Toggle menu bar`.

**Comment by xiota** [comment]
I could consider it, but would be low priority.  I am waiting on a bug fix PR for geanylua plugin before doing anything else upstream.  `geany/geany-plugins/pull/1234`

**Comment by ralf3u** [comment]
>I could consider it, but would be low priority. I am waiting on a bug fix PR for geanylua plugin before doing anything else upstream. geany/geany-plugins/pull/1234

What about to write something like this at geany-plugins 1380?

**Comment by ralf3u** [comment]
>add the lua script to the example plugins included with geanylua plugin

What do you mean with "example plugins"?

**Comment by xiota** [comment]
These geanylua script examples: https://github.com/geany/geany-plugins/tree/master/geanylua/examples

**Comment by ralf3u** [comment]
Do the geanylua script examples exist in Geany 2.0? I can't find them.

**Comment by xiota** [comment]
On Ubuntu, they're installed at `/usr/share/geany-plugins/geanylua/`

**Comment by ralf3u** [comment]
>On Ubuntu, they're installed at /usr/share/geany-plugins/geanylua/

Yes, I can see them. But how users will activate the "toggle menu bar" if it will be part of /usr/share/geany-plugins/geanylua/?

**Comment by ralf3u** [comment]
>They should be loaded by default (when geanylua plugin activated) if user hasn't created a personal geanylua config.

OK. Why not. It is less work than to program a whole plugin. It sounds simple and quite interesting. I didn't know that this was possible. The good thing is then that there is no need to copy and paste. So, users activate the plugin `Lua Scripts` in the `Plugin Manager` and then the only further step what need to be done is to define a shortcut in the Preferences in Keybindings in the Action "Lua Script" in "Toggle menu bar". Cool.

**Comment by xiota** [comment]
Mainly don't want to open bunch of PRs upstream that are never merged.  So waiting for the bugfix PR to be merged before opening another one.

**Comment by ralf3u** [comment]
>Mainly don't want to open bunch of PRs upstream that are never merged. So waiting for the bugfix PR to be merged before opening another one.

A step-by-step policy is always a good strategy. If toggle-menu-bar will be part of geanylua/examples, then it would be already a huge progress for users. Instead of building for 10 minutes the shortcut, the shortcut will be realized with some clicks.

If this month at geany-plugins 1234 will not me merged, then I intend to write during December a comment at geany-plugins 1380.

**Comment by ralf3u** [comment]
You mentioned somewhere above that you could add the `Toggle menu bar` to /usr/share/geany-plugins/geanylua/. For users it would be a great help. Could you please add the example to /usr/share/geany-plugins/geanylua/?

**Comment by xiota** [comment]
Have been waiting for geany-plugins PR 1234 to be merged.

**Comment by ralf3u** [comment]
>Have been waiting for geany-plugins PR 1234 to be merged.

Unfortunately it seems to be that that no maintainer wants to make a merge.

>I think it fixes a bug that affects one of the scripts you use.

I think that you mean the focus issue of geanylua.

**Comment by xiota** [comment]
Yes, it fixes the focus issue.  I'm waiting for it to be fixed before doing more stuff that will take a long time to be approved.

**Comment by xiota** [comment]
Hiding menu bar was added directly in Geany.  Should be available when Geany 2.2 is released.

`https://github.com/geany/geany/pull/4279`

**Comment by ralf3u** [comment]
>Hiding menu bar was added directly in Geany. Should be available when Geany 2.2 is released.

Yes, I read that some months ago at geany.

What I just recognized: Compared to the `Toggle-menu-bar` of geanylua, the Toggle menu bar in Geany 2.2 will have the feature that with F10 the menu bar will appear temporarily. I think that it is a very good idea.

But the question what I wanted to ask:
Is there an option to hide the menu bar at every startup because I read something like "Remove configurability of menubar startup visibility"?

**Comment by xiota** [comment]
I haven't looked at the code, but from previous discussion, upstream Geany devs insist on not having an option to set startup visibility.  It will be whatever the state was when Geany was last closed.

**Comment by ralf3u** [comment]
>It will be whatever the state was when Geany was last closed.

I'm OK with that.
